R/tbls.r

```diff
@@ -1,50 +1,116 @@
-#' Tibble for intervals.
+# Validity checks ---------------------------------------------------
+#' Bed-like data.frame requirements for valr functions
 #'
-#' Required column names are `chrom`, `start` and `end`.
+#' @name ivl_df
+#' @docType package
+NULL
+
+#' Bed-like data.frame requirements for valr functions
+#' @rdname ivl_df
+#' @name genome_df
+#' @docType package
+NULL
+
+#' Check bed-like data.frame requirements for valr compatibility
 #'
-#' @param x A `data_frame`
-#' @param ... params for [tibble::tibble()]
-#' @param .validate check valid column names
+#' Required column names for interval dataframes are
+#' `chrom`, `start` and `end`. Internally interval dataframes are
+#' validated using `check_interval()`
 #'
-#' @rdname tbl_interval
+#' @param x A `data.frame` or `tibble::tibble`
+#' @rdname  ivl_df
 #'
 #' @examples
+#' # using tibble
 #' x <- tibble::tribble(
 #'   ~chrom, ~start, ~end,
 #'   'chr1',  1,     50,
 #'   'chr1',  10,    75,
 #'   'chr1',  100,   120
 #' )
 #'
-#' is.tbl_interval(x)
+#' check_interval(x)
+#'
+#' # using base R data.frame
+#' x <- data.frame(chrom = "chr1",
+#'            start = 0,
+#'            end = 100,
+#'            stringsAsFactors = FALSE
+#' )
 #'
-#' x <- tbl_interval(x)
-#' is.tbl_interval(x)
+#' check_interval(x)
 #'
 #' @export
-tbl_interval <- function(x, ..., .validate = TRUE) {
+check_interval <- function(x) {
+  expect_names <- c("chrom", "start", "end")
+  check_names(x, expect_names)
+
+  if(!tibble::is_tibble(x)){
```

Comment:
You don't need the check, just coerce it.

Reply:
The check is used because coercing with `as_tibble()` will remove groups from a grouped tibble. 

``` r
library(dplyr, warn.conflicts = FALSE)
library(tibble)

x <- group_by(mtcars, cyl)
as_tibble(x)
#> # A tibble: 32 x 11
#>      mpg   cyl  disp    hp  drat    wt  qsec    vs    am  gear  carb
#>  * <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl>
#>  1  21       6  160    110  3.9   2.62  16.5     0     1     4     4
#>  2  21       6  160    110  3.9   2.88  17.0     0     1     4     4
#>  3  22.8     4  108     93  3.85  2.32  18.6     1     1     4     1
#>  4  21.4     6  258    110  3.08  3.22  19.4     1     0     3     1
#>  5  18.7     8  360    175  3.15  3.44  17.0     0     0     3     2
#>  6  18.1     6  225    105  2.76  3.46  20.2     1     0     3     1
#>  7  14.3     8  360    245  3.21  3.57  15.8     0     0     3     4
#>  8  24.4     4  147.    62  3.69  3.19  20       1     0     4     2
#>  9  22.8     4  141.    95  3.92  3.15  22.9     1     0     4     2
#> 10  19.2     6  168.   123  3.92  3.44  18.3     1     0     4     4
#> # … with 22 more rows
```

<sup>Created on 2020-04-26 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>